### PR TITLE
Moving deleteSegment call from POST to DELETE call

### PR DESCRIPTION
**Pull request by tibrewalpratik17**
label:
`bugfix`

Moving `deleteSegments` call from POST api to DELETE api like other delete-segment based apis.

### CODE REVIEW

**Review by Jackie-Jiang — COMMENTED** [review]
Shall we integrate it into `DELETE /segments/{tableName}` and add an extra parameter for target segments? If segments are not provided, we delete all segments

**Comment by tibrewalpratik17** [comment]
> Shall we integrate it into DELETE /segments/{tableName} and add an extra parameter for target segments? If segments are not provided, we delete all segments

Sounds good to me! Updated accordingly!

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java [hunk 1]
```diff
@@ -666,26 +666,38 @@ public SuccessResponse deleteSegment(
...
+  @ApiOperation(value = "Delete the list of segments provided in the payload else all segments",
+      notes = "Delete the list of segments provided in the payload else all segments")
+  public SuccessResponse deleteMultipleSegments(
       @ApiParam(value = "Name of the table", required = true) @PathParam("tableName") String tableName,
       @ApiParam(value = "OFFLINE|REALTIME", required = true) @QueryParam("type") String tableTypeStr,
       @ApiParam(value = "Retention period for the table segments (e.g. 12h, 3d); If not set, the retention period "
           + "will default to the first config that's not null: the table config, then to cluster setting, then '7d'. "
           + "Using 0d or -1d will instantly delete segments without retention")
-      @QueryParam("retention") String retentionPeriod, @Context HttpHeaders headers) {
+      @QueryParam("retention") String retentionPeriod, List<String> segments, @Context HttpHeaders headers) {
```
Comment (Jackie-Jiang) [inline pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java:677]: Do we want to take it as a payload, or a query param? Is it common for DELETE request to take payload?

Comment (tibrewalpratik17) [reply pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java:677]: Hmm good call! I don't think DELETE requests should have payloads as standard practise. Let me update!

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java [hunk 2]
```diff
@@ -666,26 +666,40 @@ public SuccessResponse deleteSegment(
...
+  public SuccessResponse deleteMultipleSegments(
       @ApiParam(value = "Name of the table", required = true) @PathParam("tableName") String tableName,
       @ApiParam(value = "OFFLINE|REALTIME", required = true) @QueryParam("type") String tableTypeStr,
       @ApiParam(value = "Retention period for the table segments (e.g. 12h, 3d); If not set, the retention period "
           + "will default to the first config that's not null: the table config, then to cluster setting, then '7d'. "
           + "Using 0d or -1d will instantly delete segments without retention")
-      @QueryParam("retention") String retentionPeriod, @Context HttpHeaders headers) {
+      @QueryParam("retention") String retentionPeriod,
+      @ApiParam(value = "Segment names to be deleted if not provided deletes all segments by default",
+          allowMultiple = true) @QueryParam("segmentNames") List<String> segments, @Context HttpHeaders headers) {
```
Comment (Jackie-Jiang) [inline pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java:679]: Suggest renaming it to be aligned with `columns` in other APIs
```suggestion
          allowMultiple = true) @QueryParam("segments") List<String> segments, @Context HttpHeaders headers) {
```

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java [hunk 3]
```diff
@@ -666,26 +666,40 @@ public SuccessResponse deleteSegment(
...
     TableType tableType = Constants.validateTableType(tableTypeStr);
     if (tableType == null) {
       throw new ControllerApplicationException(LOGGER, "Table type must not be null", Status.BAD_REQUEST);
     }
     String tableNameWithType =
         ResourceUtils.getExistingTableNamesWithType(_pinotHelixResourceManager, tableName, tableType, LOGGER).get(0);
-    deleteSegmentsInternal(tableNameWithType,
-        _pinotHelixResourceManager.getSegmentsFromPropertyStore(tableNameWithType), retentionPeriod);
-    return new SuccessResponse("All segments of table " + tableNameWithType + " deleted");
+    if (segments == null || segments.isEmpty()) {
```
Comment (Jackie-Jiang) [inline pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java:687]: (nit) I think it cannot be `null`
```suggestion
    if (segments.isEmpty()) {
```

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java [hunk 4]
```diff
@@ -666,26 +666,40 @@ public SuccessResponse deleteSegment(
   @Path("/segments/{tableName}")
   @Authorize(targetType = TargetType.TABLE, paramName = "tableName", action = Actions.Table.DELETE_SEGMENT)
   @Authenticate(AccessType.DELETE)
-  @ApiOperation(value = "Delete all segments", notes = "Delete all segments")
-  public SuccessResponse deleteAllSegments(
+  @ApiOperation(value = "Delete the list of segments provided in the payload else all segments",
```
Comment (Jackie-Jiang) [inline pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java:669]: Please also update the api doc accordingly

**Review by Jackie-Jiang — APPROVED** [review]
Suggest adding a test in `PinotSegmentRestletResourceTest` for the new usage

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java (hunk 4 above)
Comment (tibrewalpratik17) [reply pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java:669]: Already updated. Do we want to add anything else?

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java (hunk 3 above)
Comment (tibrewalpratik17) [reply pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java:687]: I see with "columns" in other APIs we do check for null as well. Let's keep this same for parity?

**Comment by tibrewalpratik17** [comment]
> Suggest adding a test in PinotSegmentRestletResourceTest for the new usage

Will do Jackie.

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java (hunk 4 above)
Comment (Jackie-Jiang) [reply pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java:669]: We no longer take payload, we take query param instead

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java (hunk 2 above)
Comment (Jackie-Jiang) [reply pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java]: ^^ (Maybe you haven't pushed the change yet)

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java (hunk 3 above)
Comment (Jackie-Jiang) [reply pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java:687]: I didn't see the `null` check in `getSegmentMetadata()`, but it doesn't harm to check anyway

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java (hunk 2 above)
Comment (tibrewalpratik17) [reply pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java]: committed this

**Pull request merged**